src/bin/pg_autoctl/cli_common.c

```diff
@@ -1478,6 +1478,66 @@ cli_common_pgsetup_init(ConfigFilePaths *pathnames, PostgresSetup *pgSetup)
 }
 
 
+/*
+ * cli_common_ensure_formation reads the formation name from the configuration
+ * file where it's not been given on the command line. When the local node is a
+ * monitor, the target formation should be found on the command line with the
+ * option --formation, otherwise we default to FORMATION_DEFAULT.
+ */
+bool
+cli_common_ensure_formation(KeeperConfig *options)
+{
+	/* if --formation has been used, we're good */
+	if (!IS_EMPTY_STRING_BUFFER(options->formation))
+	{
+		return true;
+	}
+
+	switch (ProbeConfigurationFileRole(options->pathnames.config))
+	{
+		case PG_AUTOCTL_ROLE_MONITOR:
```

Comment:
My understanding and testing shows that the monitor is always part of the default formation. So, we cannot have a monitor on a non-default formation, right?

Reply:
I am used to thinking of it in a different way. In my view the monitor is not part of any formation, it hosts one or many formations, and comes with the special formation named “default” created by the extension script, as a way to simplify the first deployment.

That's why when doing `pg_autoctl perform failover` from the monitor, a target formation needs to be picked.